MPI_Session is an MPI 4.0 handle

It appears that `MPI_Session` is used in `general/communication.hpp` as a class name. MPI 4.0 added `MPI_Session` as a standard MPI handle type. Spack E4S builds with MPICH versions >= 4.0 fail because of the ambiguity.

https://github.com/mfem/mfem/blob/168b19fc4904dc90e217bef9b5c3f595c4865da2/general/communication.hpp#L90-L104

```
ex1p.cpp(66): error: "MPI_Session" is ambiguous
1 error detected in the compilation of "ex1p.cpp".
makefile:78: recipe for target 'ex1p' failed
make: *** [ex1p] Error 1
make: Leaving directory '/tmp/root/spack-stage/spack-stage-mfem-4.3.0-eh3uxmai2x52ienouug4aiftaydipfd6/spack-src/examples'
```

--- a/miniapps/navier/navier_turbchan.cpp
+++ b/miniapps/navier/navier_turbchan.cpp
@@ -89,7 +89,8 @@ void vel_wall(const Vector &x, double t, Vector &u)
 
 int main(int argc, char *argv[])
 {
-   MPI_Session mpi(argc, argv);
+   Mpi::Init();
+   Hypre::Init();
 
    double Lx = 2.0 * M_PI;
    double Ly = 1.0;
@@ -121,7 +122,7 @@ int main(int argc, char *argv[])
    Mesh periodic_mesh = Mesh::MakePeriodic(mesh,
                                            mesh.CreatePeriodicVertexMapping(translations));
 
-   if (mpi.Root())
+   if (Mpi::Root())
    {
       printf("NL=%d NX=%d NY=%d NZ=%d dx+=%f\n", NL, NX, NY, NZ, LC * ctx.Re_tau);
       std::cout << "Number of elements: " << mesh.GetNE() << std::endl;
@@ -193,7 +194,7 @@ int main(int argc, char *argv[])
          dt = 1e-2;
       }
 
-      if (mpi.Root())
+      if (Mpi::Root())
       {
          printf("%11s %11s\n", "Time", "dt");
          printf("%.5E %.5E\n", t, dt);

